src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java

```diff
@@ -24,32 +24,214 @@
 
 package org.takes.facets.auth.social;
 
+import java.io.FileInputStream;
 import java.io.IOException;
-import org.junit.Ignore;
+import java.net.HttpURLConnection;
+import javax.json.Json;
+import nl.jqno.equalsverifier.EqualsVerifier;
+import nl.jqno.equalsverifier.Warning;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.takes.Response;
+import org.takes.facets.auth.Identity;
+import org.takes.misc.Href;
+import org.takes.rq.RqFake;
+import org.takes.rs.RsWithBody;
+import org.takes.rs.RsWithStatus;
+import org.takes.rs.RsWithType;
 
 /**
  * Test case for {@link PsLinkedin}.
  * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
  * @version $Id$
  * @since 0.11.3
+ * @checkstyle MultipleStringLiteralsCheck (500 lines)
  */
+@RunWith(MockitoJUnitRunner.class)
 public final class PsLinkedinTest {
+    /**
+     * Fake correct REQUEST.
+     */
+    public static final RqFake REQUEST = new RqFake("GET", "/?code=200");
+
+    /**
+     * Mocked member profile json object.
+     */
+    @Mock
+    private transient MemberProfileJson memberprofilejson;
+
+    /**
+     * Test set up.
+     * @throws Exception If some problem insidemo
+     */
+    @Before
+    public void setUp() throws Exception {
+        Mockito.when(
+            this.memberprofilejson.fetch(
+                Mockito.anyString(), Mockito.anyString(),
+                Mockito.any(Href.class)
+            )
+        ).thenReturn(
+                Json.createReader(
+                    new FileInputStream(
+                        PsLinkedinTest.class.getResource(
+                            "linkedinTestUser.json"
+                        ).getPath()
+                    )
+                ).readObject()
+            );
+    }
 
     /**
      * PsLinkedin can login.
      * @throws IOException If some problem inside
-     * @todo #12:30min/DEV This class not implemented yet, but has to be.
-     *  Please implement it. Mocked OAuth server is one of the possible
-     *  solutions. In this case, we could use some library like WireMock
-     *  (see http://www.wiremock.org), which allows make stub for web
-     *  service. We'll need to make some changes in PsLinkedin,
-     *  that allow to change API url, and use local stubbed http
-     *  server during unit test.
-     */
-    @Ignore
+     */
+    @Test
+    public void successfulEnter() throws IOException {
+        new PsLinkedin(this.memberprofilejson)
+            .enter(REQUEST);
+    }
+
+    /**
+     * The authorization code is not provided with the REQUEST.
```

Comment:
@ikhvostenkov generally test javadocs should be in a form: *SomeClass can do something* and test method names should be *doesSomething*.